### PR TITLE
Image zoom

### LINKED ISSUE #143: #92 Full screen image zoom by pinch
Implement so that you can zoom an image by pinching
https://www.androidhive.info/2013/09/android-fullscreen-image-slider-with-swipe-and-pinch-zoom-gestures/

### PR DESCRIPTION
**Pull request by aolsson711**
Created Pinch zoom on fullscreen images from gallery. 
However impossible to test... 
closes #143

### CODE REVIEW

**Review by giommok — CHANGES_REQUESTED** [review]
Code looks really good and really nice documentation, well done.
Anyway, I think we could add some tests. Further discussion on Discord.

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/gallery/ZoomableImageView.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+            maxTrans = 0;
+        }
+
+        if (trans < minTrans) {
+            return minTrans - trans;
+        }
+        if (trans > maxTrans) {
+            return maxTrans - trans;
+        }
+        return 0;
```
Comment (ravifrancesco) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/gallery/ZoomableImageView.java:192]:
```suggestion
        return trans < minTrans ? minTrans - trans : maxTrans - trans;
```

**Review by ravifrancesco — CHANGES_REQUESTED** [review]
Small comment but overall the code looks good. Is there a way to keep track of the image size? In case could we test the zoom feature simulating a gesture and seeing if the scale increases/decreases

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/gallery/ZoomableImageView.java (hunk 1 above)
Comment (aolsson711) [reply app/src/main/java/com/github/bgabriel998/softwaredevproject/gallery/ZoomableImageView.java:200]: Not sure this is correct. I am pretty sure that there exists cases where (for easy numbers) minTrans = 0, maxTrans = 10, trans = 7. Then we want to return 0. not 3

**Review by giommok — CHANGES_REQUESTED** [review]
Well done with the tests! 
I think we should use the upper-case convention for the constants in `ImageActivityTest`.
Would it be a problem to refactor them from Android Studio?

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/gallery/ZoomableImageView.java (hunk 1 above)
Comment (ravifrancesco) [reply app/src/main/java/com/github/bgabriel998/softwaredevproject/gallery/ZoomableImageView.java:200]: Oh yep you're right

**Review by ravifrancesco — APPROVED** [review]
Great job with the tests. I'll approve it, but consider @giommok request for the constant

**Review by giommok — APPROVED** [review]
All good!

**Pull request merged**